src/lib/pipeline/ReorderBuffer.cc

```diff
@@ -79,7 +79,9 @@ void ReorderBuffer::flush(uint64_t afterSeqId) {
       break;
     }
 
-    for (const auto& reg : uop->getDestinationRegisters()) {
+    auto destinations = uop->getDestinationRegisters();
```

Comment:
May be worth adding a small comment above the loop describing why it's done in reverse?